Samples/WinMLSamplesGallery/TransformAsync/SegmentModel.h

```diff
@@ -0,0 +1,152 @@
+#define USE_PIX
+//#define DBG
+#include <winrt/Microsoft.AI.MachineLearning.Experimental.h>
+#include <winrt/Microsoft.AI.MachineLearning.h>
+#include <Windows.AI.MachineLearning.native.h>
+#include <winrt/Windows.Graphics.DirectX.Direct3D11.h>
+#include <windows.media.core.interop.h>
+#include <winrt/Windows.Devices.Display.Core.h>
+#include <strsafe.h>
+#include <wtypes.h>
+#include <winrt/base.h>
+#include <dxgi.h>
+#include <d3d11.h>
+#include <mutex>
+#include <winrt/windows.foundation.collections.h>
+#include <winrt/Windows.Media.h>
+//#include <DXProgrammableCapture.h>
+#include "External/common.h"
+
+
+using namespace winrt::Microsoft::AI::MachineLearning;
+using namespace winrt::Microsoft::AI::MachineLearning::Experimental;
+using namespace winrt::Windows::Graphics::DirectX::Direct3D11;
+using namespace winrt::Windows::Media;
+
+// Model-agnostic helper LearningModels
+LearningModel Normalize0_1ThenZScore(long height, long width, long channels, const std::array<float, 3>& means, const std::array<float, 3>& stddev);
+LearningModel ReshapeFlatBufferToNCHW(long n, long c, long h, long w);
+LearningModel Invert(long n, long c, long h, long w);
+
+class StreamModelBase
+{
+public:
+	StreamModelBase(): 
+		m_inputVideoFrame(NULL),
+		m_outputVideoFrame(NULL),
+		m_session(NULL),
+		m_binding(NULL), 
+		m_syncStarted(FALSE)
+	{}
+	StreamModelBase(int w, int h) :
+		m_inputVideoFrame(NULL),
+		m_outputVideoFrame(NULL),
+		m_session(NULL),
+		m_binding(NULL),
+		m_syncStarted(FALSE)
+	{}
+	virtual ~StreamModelBase() {
+		if(m_session) m_session.Close();
+		if(m_binding) m_binding.Clear();
+		if (m_inputVideoFrame) m_inputVideoFrame.Close();
+		if (m_outputVideoFrame) m_outputVideoFrame.Close(); 
+	};
```

Comment:
Do these need to be explicitly called, given the destructor should clean up when the `StreamModelBase` does?

Reply:
Explicitly cleaning up does help make sure all the d3d-related fields are cleaned up properly, yes.